File: src/main/java/com/dh/ondot/member/core/exception/NotFoundHomeAddressException.java
package com.dh.ondot.member.core.exception;

import com.dh.ondot.core.exception.NotFoundException;

import static com.dh.ondot.core.exception.ErrorCode.NOT_FOUND_HOME_ADDRESS;

public class NotFoundHomeAddressException extends NotFoundException {
    public NotFoundHomeAddressException(Long memberId) {
        super(NOT_FOUND_HOME_ADDRESS.getMessage().formatted(memberId));
    }

    @Override
    public String getErrorCode() {
        return NOT_FOUND_HOME_ADDRESS.name();
    }
}

File: src/main/kotlin/com/dh/ondot/member/core/exception/NotFoundHomeAddressException.kt
package com.dh.ondot.member.core.exception

import com.dh.ondot.core.exception.ErrorCode.NOT_FOUND_HOME_ADDRESS
import com.dh.ondot.core.exception.NotFoundException

class NotFoundHomeAddressException(memberId: Long) :
    NotFoundException(NOT_FOUND_HOME_ADDRESS.message.format(memberId)) {
    override val errorCode: String get() = NOT_FOUND_HOME_ADDRESS.name
}
